### PR TITLE
Feature/HM-82-Images-New add logic in user images

### PR DESCRIPTION
**Pull request by petarbelchev**
The new Add logic in user images has been implemented.

### CODE REVIEW

File: src/Images/Images.Api/Controllers/UserImagesController.cs [hunk 1]
```diff
@@ -25,30 +27,49 @@ public class UserImagesController(
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
-        public async Task<IActionResult> AddImage(string userId, IFormFile image)
+        public async Task<IActionResult> AddImage([FromRoute] string userId, [ValidImage] IFormFile image)
```
Comment (rosteslav) [inline src/Images/Images.Api/Controllers/UserImagesController.cs]: From task: "New Add method/feature that only adds an image and returns `ImageUrl` from _imgbbService"
We don't get `userId` here!

File: src/Images/Images.Application/Attributes/ValidImageAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.ComponentModel.DataAnnotations;
+
+namespace BuildingMarket.Images.Application.Attributes
+{
+    public class ValidImageAttribute : ValidationAttribute
```
Comment (rosteslav) [inline src/Images/Images.Application/Attributes/ValidImageAttribute.cs:6]: did you test that?

File: src/Images/Images.Application/Features/Images/Commands/AddUserImage/AddUserImageCommand.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@ namespace BuildingMarket.Images.Application.Features.Images.Commands.AddUserImag
 {
     public class AddUserImageCommand : IRequest<string>
     {
-        public IFormFile FormFile { get; set; }
         public string UserId { get; set; }
```
Comment (rosteslav) [inline src/Images/Images.Application/Features/Images/Commands/AddUserImage/AddUserImageCommand.cs]: no userId

File: src/Images/Images.Application/Features/Images/Commands/AddUserImage/AddUserImageCommandHandler.cs [hunk 4]
```diff
@@ -4,30 +4,22 @@
...
-        private readonly IUserImagesRepository _repository = repository;
         private readonly IImgbbService _imgbbService = imgbbService;
 
-        public async Task<string> Handle(
-            AddUserImageCommand request,
-            CancellationToken cancellationToken)
+        public async Task<string> Handle(AddUserImageCommand request, CancellationToken cancellationToken)
         {
             string ext = Path.GetExtension(request.FormFile.FileName);
             var imageName = $"{request.UserId}-{Guid.NewGuid()}{ext}";
```
Comment (rosteslav) [inline src/Images/Images.Application/Features/Images/Commands/AddUserImage/AddUserImageCommandHandler.cs]: just "UserImg" instaed {request.UserId} for users

**Review by rosteslav — CHANGES_REQUESTED** [review]

File: src/Images/Images.Application/Attributes/ValidImageAttribute.cs (hunk 2 above)
Comment (petarbelchev) [reply src/Images/Images.Application/Attributes/ValidImageAttribute.cs:6]: I think so

**Review by rosteslav — APPROVED** [review]

**Pull request merged**